rasa/core/agent.py

```diff
@@ -115,13 +115,16 @@ def _load_domain_and_policy_ensemble(
 
 def _load_and_set_updated_model(
     agent: "Agent", model_directory: Text, fingerprint: Text
-) -> None:
+) -> bool:
```

Comment:
one more thought: Should we delete `core_path`, `nlu_path` in here when an exception occurs?

Reply:
See the other comment - since `model_directory` is being received as an argument, I think it is better to leave the management of the directory to the caller of this function.

Reply:
ah. I also overlooked that both directories are within `model_directory` 👍